src/lib/services/telegram/TelegramService.ts

```diff
@@ -4,28 +4,37 @@ import AuthAPI from '@/lib/api/auth/AuthAPI';
 import StorageUtil from '@/lib/utils/StorageUtil';
 import { TelegramUser } from '@/types/telegram';
 
+declare global {
+  interface Window {
+    Telegram: {
+      Login: {
+        auth: (
+          botId: string,
+          requestAccess: boolean,
+        ) => Promise<TelegramUser | null>;
+      };
+    };
+  }
+}
+
 class TelegramService {
-  private static openAuthenticationDialog() {
+  private static async openAuthenticationDialog(): Promise<TelegramUser | null> {
+    const Telegram = window.Telegram;
+    const botId = process.env.NEXT_PUBLIC_BOT_ID || '';
```

Comment:
How this is supposed to work in browser (browsers doesn't have process var)

Reply:
browser dont have process var, but environment variables such as process.env are replaced at build time, and their values are embedded directly in the javascript sent to the client, did i misunderstand something?

Reply:
k makes more sense